packages/keyring-controller/src/KeyringController.ts

```diff
@@ -640,7 +644,7 @@ export class KeyringController extends BaseController<
    * @returns Promise resolving to the added account address.
    */
   async addNewAccountWithoutUpdate(): Promise<string> {
-    return this.#withControllerLock(async () => {
+    return this.#persistOrRollback(async () => {
```

Comment:
There is still a call to `this.#updateVault` in this function - can that be removed as redundant?

Reply:
Good catch! it needs to be removed. [4a15e93](https://github.com/MetaMask/core/pull/4192/commits/4a15e936ae66114677a3f98a69bb5493aff092d2)